spec/support/corpus.rb

```diff
@@ -129,33 +129,61 @@ def checkout
       # @param path [Pathname] path responsible for exception
       #
       # @return [Integer] mutations generated
-      def count_mutations_and_check_errors(path)
+      def check_generation(path)
         relative_path = path.relative_path_from(repo_path)
 
-        count = count_mutations(path)
+        node = Parser::CurrentRuby.parse(path.read)
+        fail "Cannot parse: #{path}" unless node
+
+        mutations = Mutant::Mutator.mutate(node)
+
+        mutations.each do |mutation|
+          check_generation_invariants(node, mutation)
+        end
 
         expected_errors.assert_success(relative_path)
 
-        count
+        mutations.length
       rescue Exception => exception # rubocop:disable Lint/RescueException
         expected_errors.assert_error(relative_path, exception)
 
         DEFAULT_MUTATION_COUNT
       end
 
+      # Check generation invariants
+      #
+      # @param [Parser::AST::Node] original
+      # @param [Parser::AST::Node] mutation
+      #
+      # @return [undefined]
+      #
+      # @raise [Exception]
+      def check_generation_invariants(original, mutation)
+        return unless ENV['MUTANT_CORPUS_EXPENSIVE']
```

Comment:
Doing this per mutation is to slow. And having an ENV based flip sucks. This all hints to create a first class verification CLI.

Also cannot wait for my 2990WX to ship so I can parallelize this much more.